flows/actions/send_msg.go

```diff
@@ -161,15 +162,19 @@ func (a *SendMsgAction) getTemplateMsg(run flows.Run, urn urns.URN, channelRef *
 	var previewParts []string
 	var previewQRs []string
 
+	variables := translation.Variables()
+
 	for _, comp := range translation.Components() {
+		varMap := comp.Variables()
 		paramValues := evaluatedParams[comp.Name()]
-		params := make([]flows.TemplatingParam, len(comp.Params()))
+		params := make([]flows.TemplatingParam, len(varMap))
 
-		for i, p := range comp.Params() {
+		for i, varName := range utils.SortedKeys(varMap) {
+			v := variables[varMap[varName]]
 			if i < len(paramValues) {
-				params[i] = flows.TemplatingParam{Type: p.Type(), Value: paramValues[i]}
+				params[i] = flows.TemplatingParam{Type: v.Type(), Value: paramValues[i]}
```

Comment:
Any reason to not add `varName` in the `TemplatingParam`?
That is the new data we need to be able to support the Twilio templates which triggered this refactoring.
I guess that could be another PR too if this is just for backward compatibility